interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -529,6 +624,7 @@ func (pdb *PromptDB) Close() error {
 	// not want to send {"resolved": "cancelled"} in the notice data.
 	data := map[string]string{"resolved": "cancelled"}
 	for user, userEntry := range pdb.perUser {
+		userEntry.expirationTimer.Stop()
```

Comment:
How is this synchronized with the timer elapsing and sending another reply? I wonder if we should have some sort of timer mutex so that we only send one type of reply - either cancelled or expired but not both.

Reply:
This is synchronized by the maxIDMmap being closed with the lock held, and the callback checking whether it has been closed, again with the lock held. I'm about to push a change which removes the `expirationTimer.Stop()` entirely to since it's racy, unnecessary, and confusing.